File: benchmarks/pandas/bench_ewm_corr.py
"""Benchmark: Series.ewm(span=10).corr(other) on 100k-element Series."""
import json, time
import numpy as np
import pandas as pd

SIZE = 100_000
WARMUP = 3
ITERATIONS = 10

idx = np.arange(SIZE)
a = pd.Series(np.sin(idx * 0.01))
b = pd.Series(np.cos(idx * 0.01))
for _ in range(WARMUP): a.ewm(span=10).corr(b)

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    a.ewm(span=10).corr(b)
    times.append(time.perf_counter() - t0)
total = sum(times) * 1000
print(json.dumps({ "function": "ewm_corr", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))

File: benchmarks/tsb/bench_ewm_corr.ts
/**
 * Benchmark: EWM.corr(other) on two 100k-element Series.
 */
import { Series, EWM } from "../../src/index.js";

const SIZE = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const a = new Series({ data: Array.from({ length: SIZE }, (_, i) => Math.sin(i * 0.01)) });
const b = new Series({ data: Array.from({ length: SIZE }, (_, i) => Math.cos(i * 0.01)) });
const ewmA = new EWM(a, { span: 10 });

for (let i = 0; i < WARMUP; i++) ewmA.corr(b);

const times: number[] = [];
for (let i = 0; i < ITERATIONS; i++) {
  const t0 = performance.now();
  ewmA.corr(b);
  times.push(performance.now() - t0);
}
const total = times.reduce((a, b) => a + b, 0);
console.log(JSON.stringify({ function: "ewm_corr", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));
